gms/gossiper.cc

```diff
@@ -1206,8 +1227,14 @@ version_type gossiper::get_max_endpoint_state_version(const endpoint_state& stat
 future<> gossiper::evict_from_membership(inet_address endpoint, permit_id pid) {
     verify_permit(endpoint, pid);
     co_await mutate_live_and_unreachable_endpoints([endpoint] (gossiper& g) {
-        g._unreachable_endpoints.erase(endpoint);
-        g._live_endpoints.erase(endpoint);
+        return live_and_unreachable_endpoints{
+            boost::copy_range<std::unordered_set<inet_address>>(g._live_endpoints | boost::adaptors::filtered([endpoint] (const auto& node) {
+                return node != endpoint;
+            })),
+            boost::copy_range<std::unordered_map<inet_address, clk::time_point>>(g._unreachable_endpoints | boost::adaptors::filtered([endpoint] (const auto& node) {
+                return node.first != endpoint;
+            }))
+        };
```

Comment:
Another case where a copy provided by mutate() would improve readability.

Reply:
done in v2